src/compiler/mod.rs

```diff
@@ -0,0 +1,65 @@
+//! The compiler for `ronkathon` circuits.
+
+// TODO: Remove this once the module is fleshed out.
+#![allow(missing_docs)]
+use petgraph::graph::{DiGraph, NodeIndex};
+
+// TODO: Goal, allow someone to use this library to write rust to generate arithmetic circuits
+// basically.
+// ```
+// use ronkathon::compiler::*;
+//
+// fn main() {
+//   let a: Input = Variable::public(7);
+//   let b: Input = Variable::private(3);
+//   let b: Expression = a * c;
+// }
+// ```
+// So we can do things like `impl Add`, `impl Mul` for variables and make them into gates?
+
+// Above seems done. Now we need to have a way to unravel a collection of expressions into a
+// circuit that may have the same inputs and outputs as the expressions. Inputs are going to be
+// the terminal variables found by fully unravelling expressions and they should be named. The
+// fully ravelled expressions are the outputs, and they can also be named
+use super::*;
+
+pub mod dsl;
+
+pub struct Wire {
```

Comment:
how will wires have input and output? maybe a better name could be `Constraint`?

Reply:
oh yeah, that is better. I think I was trippin'

Reply:
Oh, got it. I got confused then. I remember previously it was names as `Connection` Maybe that made more sense? IDK, i'll let you cook.